src/app/modules/main/chat_section/module.nim

```diff
@@ -307,6 +321,62 @@ proc initContactRequestsModel(self: Module) =
 
   self.view.contactRequestsModel().addItems(contactsWhoAddedMe)
 
+proc rebuildCommunityTokenPermissionsModel(self: Module) =
+  let community = self.controller.getMyCommunity()
+  var tokenPermissionsItems: seq[TokenPermissionItem] = @[]
+  var allTokenRequirementsMet = true
+
+  for id, tokenPermission in community.tokenPermissions:
+    let tokenPermissionItem = self.buildTokenPermissionItem(tokenPermission)
+
+    if not tokenPermissionItem.tokenCriteriaMet:
+        allTokenRequirementsMet = false
+
+    tokenPermissionsItems.add(tokenPermissionItem)
+
+  self.view.tokenPermissionsModel().setItems(tokenPermissionsItems)
+  self.view.setAllTokenRequirementsMet(allTokenRequirementsMet)
+
+proc initCommunityTokenPermissionsModel(self: Module) =
+  self.rebuildCommunityTokenPermissionsModel()
+
+proc buildTokenList(self: Module) =
+  
+  var tokenListItems: seq[TokenListItem]
+  var collectiblesListItems: seq[TokenListItem]
+
+  let erc20Tokens = self.controller.getTokenList()
+  let communityTokens = self.controller.getCommunityTokenList()
+
+  for token in erc20Tokens:
+    let tokenListItem = initTokenListItem(
+      token.symbol,
+      token.name,
+      token.symbol,
+      token.color,
+      "",
```

Comment:
Could you add the arg name before for clarity? thanks

Reply:
Will do!